CRM/Civicase/Event/Listener/CaseTypeCategoryIsActiveToggler.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+use CRM_Civicase_Factory_CaseTypeCategoryEventHandler as CaseTypeCategoryEventHandlerFactory;
+use Civi\API\Event\RespondEvent;
+
+/**
+ * Class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler.
+ *
+ * Contains extra processing for case type category when it is
+ * enabled / disabled, e.g. show / hide related menu item.
+ */
+class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler {
+
+  /**
+   * Update case type category related objects on category update.
+   *
+   * @param \Civi\API\Event\RespondEvent $e
+   *   Event data.
+   */
+  public static function onRespond(RespondEvent $e) {
+    $apiRequest = $e->getApiRequest();
+    if ($apiRequest['entity'] != 'OptionValue' || $apiRequest['action'] != 'create') {
+      return;
+    }
+    if (!self::checkReferrer() || !self::checkOptionGroup($apiRequest['params']['option_group_id'])) {
```

Comment:
Will the option group ID always be passed when updating an option value? I don't think so. I checked the ticket and the code snippet you pasted about this API call 
```php
{entity: option_value, action: create, json: {"id":"977","is_active":0}}
```
As seen above, the ID passed is the option value ID and not the option group ID.
Looks like the best bet is to use the option value ID itself to confirm that it belongs to the case type category option group.
Here's a quick code snippet for that
```php
$result = civicrm_api3('OptionValue', 'get', [
  'return' => ["option_group_id.name"],
  'id' => 5,
]);
```